pkg/resourcecollector/persistentvolumeclaim.go

```diff
@@ -101,11 +102,23 @@ func (r *ResourceCollector) preparePVCResourceForApply(
 		// In the case of storageClassMappings, we need to reset the
 		// storage class annotation and the provisioner annotation
 		var newSc string
+		var currentSc string
 		var exists bool
 		var provisioner string
-		if val, ok := pvc.Annotations[v1.BetaStorageClassAnnotation]; ok {
-			if newSc, exists = storageClassMappings[val]; exists && len(newSc) > 0 {
-				pvc.Annotations[v1.BetaStorageClassAnnotation] = newSc
+		// Get the existing storage class from the pvc spec
+		// It can be in BetaStorageClassAnnotation annotation or in the spec.
+		currentSc, err := utils.GetStorageClassNameForPVC(&pvc)
+		if err != nil {
+			return false, err
+		}
+		if len(currentSc) != 0 {
+			if newSc, exists = storageClassMappings[currentSc]; exists && len(newSc) > 0 {
+				if _, ok := pvc.Annotations[v1.BetaStorageClassAnnotation]; ok {
+					pvc.Annotations[v1.BetaStorageClassAnnotation] = newSc
+				}
+				if pvc.Spec.StorageClassName != nil && len(*pvc.Spec.StorageClassName) > 0 {
+					*pvc.Spec.StorageClassName = newSc
```

Comment:
Should we always set the storage class regardless of whether it was present on the original PVC ?

Reply:
This path will be executed only when storageclass mapping is present. Also we do not reset the PVC spec's storage class during backup, that means old storageclass will be there always.